demos/common/cpp/utils/src/images_capture.cpp

```diff
@@ -163,15 +167,26 @@ class VideoCapWrapper : public ImagesCapture {
                 nextImgId = 1;
                 cv::Mat img;
                 cap.read(img);
+                if (safe_read) {
+                    img = img.clone();
+                }
                 readerMetrics.update(startTime);
                 return img;
             }
             return cv::Mat{};
         }
         cv::Mat img;
-        if (!cap.read(img) && loop && cap.set(cv::CAP_PROP_POS_FRAMES, initialImageId)) {
+        bool success = cap.read(img);
```

Comment:
There should also be the image cloning in case of `safe_read` after first frame reading, isn't it?

Reply:
It's below: `img = img.clone();`

Reply:
It's cloned if the condition is satisfied and we reach the end of the video

Reply:
Updated